graph/memstore/quadstore.go

```diff
@@ -178,7 +185,10 @@ func (qs *QuadStore) addPrimitive(p *Primitive) int64 {
 }
 
 func (qs *QuadStore) appendPrimitive(p *Primitive) {
+	qs.prw.Lock()
 	qs.prim[p.ID] = p
+	qs.prw.Unlock()
+
 	if !qs.reading {
 		qs.all = append(qs.all, p)
```

Comment:
This will cause races as well.

Reply:
ok, I'll add a multi-thread add quad test case